app/tests/hanging_protocols_tests/test_forms.py

```diff
@@ -102,3 +102,47 @@ def test_hanging_protocol_form(client):
         "Each viewport can only be used once." not in response.rendered_content
     )
     assert HangingProtocol.objects.count() == 1
+
+
+@pytest.mark.django_db
+def test_hanging_protocol_dimension_validation(client):
+    user = UserFactory()
+    assign_perm("hanging_protocols.add_hangingprotocol", user)
+
+    response = get_view_for_user(
+        viewname="hanging-protocols:create",
+        client=client,
+        method=client.post,
+        data={
+            "title": "main",
+            "json": '[{"viewport_name": "main", "x": 0, "y": 0}]',
+        },
+        follow=True,
+        user=user,
+    )
+
+    assert response.status_code == 200
+    assert HangingProtocol.objects.count() == 0
+    assert (
+        "Viewport definition must contain all or none of x, y, w, and h"
+        in response.rendered_content
+    )
+
+    response = get_view_for_user(
+        viewname="hanging-protocols:create",
+        client=client,
+        method=client.post,
+        data={
+            "title": "main",
+            "json": '[{"viewport_name": "main", "x": 0, "y": 0, "w": 1, "h": 1}]',
+        },
+        follow=True,
+        user=user,
+    )
+
+    assert response.status_code == 200
+    assert HangingProtocol.objects.count() == 1
+    assert (
+        "Viewport definition must contain all or none of x, y, w, and h"
+        not in response.rendered_content
+    )
```

Comment:
Maybe add a test for the case where there are multiple viewports and one contains all of the dims and the other none resulting in an error.